src/visitors/hoist.js

```diff
@@ -183,122 +182,201 @@ module.exports = {
       !path.scope.hasBinding('require');
 
     if (isRequireResolve) {
-      let name =
-        '$' + asset.id + '$require_resolve$' + t.toIdentifier(args[0].value);
-      path.replaceWith(t.identifier(name));
-    }
-  },
-  ExportAllDeclaration(path, asset) {
-    if (BREAK_COMMON_JS) {
-      path.replaceWith(
-        t.identifier(
-          '$' +
-            asset.id +
-            '$expand_exports$' +
-            t.toIdentifier(path.node.source.value)
-        )
-      );
-    } else {
-      path.replaceWith(
-        EXPORT_ALL_TEMPLATE({
-          EXPORTS: t.identifier('$' + asset.id + '$exports'),
-          SOURCE: t.identifier(
-            '$' +
-              asset.id +
-              '$require$' +
-              t.toIdentifier(path.node.source.value)
-          )
-        })
-      );
+      path.replaceWith(getIdentifier(asset, 'require_resolve', args[0].value));
     }
   },
+
   ImportDeclaration(path, asset) {
-    let {source, specifiers} = path.node;
-
-    if (source && specifiers.length > 0) {
-      if (BREAK_COMMON_JS) {
-        specifiers.forEach(specifier =>
-          path.scope.rename(
-            specifier.local.name,
-            '$' +
-              asset.id +
-              '$named_import$' +
-              t.toIdentifier(source.value) +
-              '$' +
-              specifier.imported.name
-          )
+    // For each specifier, rename the local variables to point to the imported name.
+    // This will be replaced by the final variable name of the resolved asset in the packager.
+    for (let specifier of path.node.specifiers) {
+      if (t.isImportDefaultSpecifier(specifier)) {
+        path.scope.rename(
+          specifier.local.name,
+          getName(asset, 'import', path.node.source.value, 'default')
         );
-        path.remove();
-      } else {
-        path.replaceWith(
-          t.variableDeclaration(
-            'var',
-            specifiers.map(specifier =>
-              t.variableDeclarator(
-                specifier.local,
-                t.memberExpression(
-                  t.identifier(
-                    '$' + asset.id + '$require$' + t.toIdentifier(source.value)
-                  ),
-                  specifier.imported
-                )
-              )
-            )
+      } else if (t.isImportSpecifier(specifier)) {
+        path.scope.rename(
+          specifier.local.name,
+          getName(
+            asset,
+            'import',
+            path.node.source.value,
+            specifier.imported.name
           )
         );
+      } else if (t.isImportNamespaceSpecifier(specifier)) {
+        path.scope.rename(
```

Comment:
importing a namespace e.g. `import * from 'blah'` just does the same thing as a commonjs require.

Reply:
for stuff like this I thought about a Babel transform we could do in the worker, where basically we would replaced wildcard imports to named if possible (bailout if the namespace if called). We could also do the same to turn for example `const {foo} = require('foo')` to `import {foo} from 'foo'`, it may simplify our logic an should be safe enough

Reply:
Yeah that could be done. I wonder if minifiers will be able to do that for us though. We're already creating an object with the namespace for every module, so it seems like the minifier should be able to statically resolve those names off the object...